packages/app/src/components/VisualizationTypeSelector/VisualizationTypeMenuItem.js

```diff
@@ -1,38 +1,36 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import MenuItem from '@material-ui/core/MenuItem/MenuItem';
-import { visTypeDisplayNames } from '@dhis2/analytics';
 
 import ListItemIcon from '@material-ui/core/ListItemIcon/ListItemIcon';
 import ListItemText from '@material-ui/core/ListItemText/ListItemText';
-import VisualizationTypeIcon from './VisualizationTypeIcon';
+import MenuItemIcon from './MenuItemIcon';
 
 const VisualizationTypeMenuItem = ({
-    type,
-    visualizationType,
+    iconType,
     styles,
+    label,
+    isSelected,
     ...props
 }) => (
     <MenuItem
-        selected={type === visualizationType}
+        selected={Boolean(isSelected)}
         style={styles.menuItem}
         disableRipple
         {...props}
     >
         <ListItemIcon style={styles.listItemIcon}>
-            <VisualizationTypeIcon type={type} style={styles.listItemSvg} />
+            <MenuItemIcon iconType={iconType} style={styles.listItemSvg} />
         </ListItemIcon>
         <ListItemText
-            primary={visTypeDisplayNames[type]}
+            primary={label}
             disableTypography={true}
             style={styles.listItemText}
         />
     </MenuItem>
 );
 
 VisualizationTypeMenuItem.propTypes = {
-    type: PropTypes.oneOf(Object.keys(visTypeDisplayNames)),
-    visualizationType: PropTypes.oneOf(Object.keys(visTypeDisplayNames)),
     styles: PropTypes.object,
 };
```

Comment:
Add `iconType`, `label`, `isSelected` prop types

Reply:
And what about `onClick` and `disabled`? I see those are passed along to MenuItem, so in that case I'm not sure where the prop type should be added...

Reply:
As they're only getting sent through to the MUI component without any interference prop types would be redundant / unnecessary, since the component would be agnostic to the fact. Otherwise we'd need to specify all MUI props available.